### PR TITLE
[BUG] Bump up max_header_size

### PR DESCRIPTION
**Pull request by raunakab**
# Overview
Bump up the `MAX_HEADER_SIZE` constant that's being used to compare against the page sizes.

**(Naming seems weird. We're using the max "header" size to guard against "page" sizes.)**

### CODE REVIEW

File: src/daft-parquet/src/file.rs [hunk 1]
```diff
@@ -316,7 +316,7 @@ impl ParquetFileReader {
     // Set to a very high number 256MB to guard against unbounded large
     // downloads from remote storage, which likely indicates corrupted Parquet data
     // See: https://github.com/Eventual-Inc/Daft/issues/1551
-    const MAX_HEADER_SIZE: usize = 256 * 1024 * 1024;
+    const MAX_HEADER_SIZE: usize = 2 * 1024 * 1024 * 1024;
```
Comment (desmondcheongzx) [inline src/daft-parquet/src/file.rs:319]: Could we update the comment above this to explain our rationale for using 2GiB?

Comment (raunakab) [reply src/daft-parquet/src/file.rs]: Yessir! 🚀🚀🚀

**Review by desmondcheongzx — APPROVED** [review]
Looks good! Send it

**Comment by jaychia** [comment]
Hey what's happening with this PR? Is this good to merge?

**Comment by raunakab** [comment]
> Hey what's happening with this PR? Is this good to merge?

Ingress works, but currently debugging egress with PyArrow.

**Pull request merged**